# Indonesian Strategic Food Prices: Procurement Risk Analysis (2022–2026)

**Business question.** A restaurant chain operating in Java asks:
> *Which ingredients carry the most price risk, when does that risk appear, and where could we source them more cheaply?*

**Data.** Daily retail prices of 10 strategic food commodities in traditional markets across 34 Indonesian provinces, Jan 2022 – Feb 2026. Source: Bank Indonesia's PIHPS, via [Kaggle](https://www.kaggle.com/datasets/muhyusuf1112/indonesia-commodity-price-based-piphs-source).

**Stack.** Python (pandas) to load and reshape → PostgreSQL for all analysis (queries live in `sql/`) → Power BI dashboard.

**How to run**
1. Download the dataset into `data/pasar_tradisional/Cleaned_After_Imputation/`
2. Start the database: `docker compose up -d`
3. Run the cells top to bottom

**Contents**
1. Setup
2. Raw data structure
3. Reshape: wide → long
4. Data quality checks
5. Load into PostgreSQL
6. Analysis (Q1–Q4)
7. Recommendations & limitations
8. *(Optional)* Publish to Neon for Power BI

## 1. Setup

`ROOT` points to the project root whether the notebook is started from `notebooks/` or from the root.
The database URL defaults to the local Docker container defined in `docker-compose.yml`.

In [18]:
from pathlib import Path
import os
import re

import pandas as pd
from sqlalchemy import create_engine, text

ROOT = Path.cwd() if (Path.cwd() / "sql").exists() else Path.cwd().parent
DATA_DIR = ROOT / "data" / "pasar_tradisional" / "Cleaned_After_Imputation"

DB_URL = os.getenv("DB_URL", "postgresql+psycopg2://analyst:analyst@localhost:5434/harga_pangan")
engine = create_engine(DB_URL)

## 2. Raw data structure

The dataset ships as **one CSV per commodity** (10 files). Each file is in **wide format**: one row per date, one column per province (34 columns).

In [19]:
files = sorted(DATA_DIR.glob("*.csv"))
print(f"{len(files)} files found, e.g. {files[0].name}")

sample = pd.read_csv(files[0])
print("shape     :", sample.shape)
print("date range:", sample["Date_Param"].min(), "→", sample["Date_Param"].max())
print("nulls     :", int(sample.isna().sum().sum()))
sample.head(3)

10 files found, e.g. final_komoditas_tradisional_bawang_merah_2022_2026.csv
shape     : (1415, 35)
date range: 2022-01-01 → 2026-02-12
nulls     : 0


,Date_Param,Aceh,Bali,Banten,Bengkulu,DI Yogyakarta,DKI Jakarta,Gorontalo,Jambi,Jawa Barat,...,Papua Barat,Riau,Sulawesi Barat,Sulawesi Selatan,Sulawesi Tengah,Sulawesi Tenggara,Sulawesi Utara,Sumatera Barat,Sumatera Selatan,Sumatera Utara
0,2022-01-01,32000.0,22000.0,27400.0,30000.0,26000.0,34150.0,35850.0,24500.0,27350.0,...,46250.0,28400.0,29250.0,25150.0,30750.0,32150.0,33400.0,26250.0,31000.0,30150.0
1,2022-01-02,32000.0,22000.0,27400.0,30000.0,26000.0,34150.0,35850.0,24500.0,27350.0,...,46250.0,28400.0,29250.0,25150.0,30750.0,32150.0,33400.0,26250.0,31000.0,30150.0
2,2022-01-03,32100.0,21750.0,27250.0,29750.0,26250.0,32500.0,38150.0,24500.0,27300.0,...,46250.0,28450.0,24350.0,25200.0,33250.0,32150.0,33400.0,26250.0,31250.0,30500.0


**Why this format is a problem.** In wide format, the commodity lives in the *file name* and the province lives in the *column name*. A question like *"which commodity is most volatile?"* can't be answered with a single `GROUP BY`. SQL works best when every dimension is a **column value**, not a column name.

## 3. Reshape: wide → long

Target grain: **one row = one date × one province × one commodity**:

```
WIDE                                  LONG
Date_Param | Aceh | Bali | ...   →    tanggal    | provinsi | komoditas    | harga
2022-01-01 | 32000| 22000| ...        2022-01-01 | Aceh     | bawang_merah | 32000
```

`melt` unpivots the 34 province columns into two columns (`provinsi`, `harga`); the commodity name is extracted from the file name.

In [20]:
frames = []
for f in files:
    # "final_komoditas_tradisional_bawang_merah_2022_2026.csv" → "bawang_merah"
    komoditas = re.sub(r"^final_komoditas_tradisional_|_\d{4}_\d{4}$", "", f.stem)

    long = pd.read_csv(f).melt(id_vars="Date_Param", var_name="provinsi", value_name="harga")
    long["komoditas"] = komoditas
    frames.append(long)

harga = (
    pd.concat(frames, ignore_index=True)
      .rename(columns={"Date_Param": "tanggal"})
)
harga["tanggal"] = pd.to_datetime(harga["tanggal"], format="%Y-%m-%d")

print("shape      :", harga.shape)
print("commodities:", sorted(harga["komoditas"].unique()))
harga.head()

shape      : (481236, 4)
commodities: ['bawang_merah', 'bawang_putih', 'beras', 'cabai_merah', 'cabai_rawit', 'daging_ayam', 'daging_sapi', 'gula_pasir', 'minyak_goreng', 'telur_ayam']


,tanggal,provinsi,harga,komoditas
0,2022-01-01,Aceh,32000.0,bawang_merah
1,2022-01-02,Aceh,32000.0,bawang_merah
2,2022-01-03,Aceh,32100.0,bawang_merah
3,2022-01-04,Aceh,31650.0,bawang_merah
4,2022-01-05,Aceh,31000.0,bawang_merah


## 4. Data quality checks

### 4.1 Grain & value checks
If the grain is complete, the row count should equal *unique dates × 34 provinces × 10 commodities*.

In [21]:
n_days = harga["tanggal"].nunique()
expected = n_days * 34 * 10

print(f"unique dates : {n_days} ({harga['tanggal'].min().date()} → {harga['tanggal'].max().date()})")
print(f"rows         : {len(harga):,} (expected {expected:,}, gap {expected - len(harga):,})")
print(f"duplicates   : {harga.duplicated(['tanggal', 'provinsi', 'komoditas']).sum()}")
print(f"price <= 0   : {(harga['harga'] <= 0).sum()}")
print(f"nulls        : {harga['harga'].isna().sum()}")

unique dates : 1417 (2022-01-01 → 2026-02-12)
rows         : 481,236 (expected 481,780, gap 544)
duplicates   : 0
price <= 0   : 0
nulls        : 0


### 4.2 Where do the gaps come from?
Two checks: calendar days missing from **every** file, and commodities that have **fewer dates** than others.

In [22]:
all_days = pd.date_range(harga["tanggal"].min(), harga["tanggal"].max())
missing = all_days.difference(harga["tanggal"].unique())

print(f"{len(missing)} of {len(all_days)} calendar days missing ({len(missing) / len(all_days):.1%})")
print("by weekday      :", pd.Series(missing.day_name()).value_counts().to_dict())
print("missing weekdays:", [d.strftime("%Y-%m-%d (%a)") for d in missing if d.weekday() < 5])

days_per_commodity = harga.groupby("komoditas")["tanggal"].nunique()
print("\ndates per commodity:")
print(days_per_commodity.sort_values())

87 of 1504 calendar days missing (5.8%)
by weekday      : {'Sunday': 49, 'Saturday': 33, 'Thursday': 2, 'Monday': 1, 'Tuesday': 1, 'Wednesday': 1}
missing weekdays: ['2022-03-03 (Thu)', '2022-05-02 (Mon)', '2022-05-03 (Tue)', '2022-05-26 (Thu)', '2022-06-01 (Wed)']

dates per commodity:
komoditas
bawang_merah     1415
bawang_putih     1415
beras            1415
daging_ayam      1415
gula_pasir       1415
daging_sapi      1415
telur_ayam       1415
minyak_goreng    1415
cabai_rawit      1417
cabai_merah      1417
Name: tanggal, dtype: int64


### Data quality findings

| # | Finding | Impact |
|---|---|---|
| 1 | Source data was **pre-imputed** by the uploader (missing cells filled; method unknown). | Volatility may be understated. Stated as a limitation. |
| 2 | **87 of 1,504 calendar days (5.8%)** are absent from all files, mostly sporadic weekends, plus 2022 national holidays (Nyepi, Idulfitri, Ascension Day, Pancasila Day). Imputation filled empty cells but not missing rows. | `LAG()` on daily rows can span several days → volatility and trends are analysed at **monthly / yearly** level. |
| 3 | Cabai merah & cabai rawit have **2 extra dates** (1,417 vs 1,415). 8 commodities × 2 dates × 34 provinces = the 544-row gap above. | Negligible after monthly aggregation. |
| 4 | No duplicates at grain, no zero/negative prices, no nulls. | Grain is enforced as a primary key in PostgreSQL (next step). |

Lebaran windows from 2023 onwards contain no gaps, so the Lebaran analysis (Q2) is not affected by finding 2.

## 5. Load into PostgreSQL

All analysis runs in SQL, so the long table is loaded once into Postgres. The **primary key on `(tanggal, provinsi, komoditas)`** is the database-level grain test: it fails if any duplicate exists.

In [23]:
harga.to_sql(
    "harga_pangan", engine,
    if_exists="replace",   # drop & recreate → safe to re-run
    index=False,           # don't store the pandas index as a column
    chunksize=10_000,
    method="multi",        # many rows per INSERT → much faster
)

with engine.begin() as conn:  # begin() commits automatically on success
    conn.execute(text("ALTER TABLE harga_pangan ALTER COLUMN tanggal TYPE date"))
    conn.execute(text("ALTER TABLE harga_pangan ADD PRIMARY KEY (tanggal, provinsi, komoditas)"))

pd.read_sql("SELECT komoditas, COUNT(*) AS n FROM harga_pangan GROUP BY komoditas ORDER BY komoditas", engine)

,komoditas,n
0,bawang_merah,48110
1,bawang_putih,48110
2,beras,48110
3,cabai_merah,48178
4,cabai_rawit,48178
5,daging_ayam,48110
6,daging_sapi,48110
7,gula_pasir,48110
8,minyak_goreng,48110
9,telur_ayam,48110


## 6. Analysis

Each question lives in its own file in `sql/` (one file = one question = one statement), so the SQL can be read directly on GitHub. The helper below reads a file and returns the result as a DataFrame.

In [24]:
def run_sql(path: str) -> pd.DataFrame:
    """Read a .sql file (path relative to the project root) and run it against Postgres."""
    return pd.read_sql(text((ROOT / path).read_text()), engine)

### Q1. Which commodities are most volatile?

**Approach.** Aggregate to a national monthly average first (removes the date-gap problem), then measure month-over-month change with `LAG() OVER (PARTITION BY komoditas ORDER BY bulan)`. The main metric is the **average absolute monthly change**; the coefficient of variation (CV) is shown for comparison.

*SQL: CTEs, `LAG` window function, `DATE_TRUNC`.*

In [25]:
run_sql("sql/01_volatilitas.sql")

,komoditas,rata_gejolak_bulanan_pct,lonjakan_terbesar_pct,penurunan_terbesar_pct,cv_pct
0,cabai_rawit,14.3,58.2,-28.6,18.8
1,cabai_merah,11.7,42.2,-28.1,19.5
2,bawang_merah,11.2,32.2,-29.1,18.0
3,bawang_putih,3.1,10.4,-6.9,15.7
4,daging_ayam,2.7,7.0,-7.3,4.6
5,telur_ayam,2.5,8.7,-13.3,5.5
6,minyak_goreng,1.6,17.7,-7.8,6.7
7,beras,0.9,3.9,-1.7,10.8
8,gula_pasir,0.9,5.1,-0.9,10.4
9,daging_sapi,0.4,3.2,-0.7,2.0


**Findings**
- **Three risk tiers.** Fresh spices (cabai rawit, cabai merah, bawang merah) swing **11–14% per month** on average; cabai rawit once jumped **58% in a single month**. Bawang putih, chicken and eggs sit around 2.5–3%. Oil, rice, sugar and beef are below 2%.
- **Volatility ≠ trend.** Beras and gula barely move month to month (0.9%) yet have a CV of ~10%. CV measures spread over the whole period, so a *slow, steady increase* inflates it. These two carry **trend risk**, not volatility risk. Q4 confirms this.

### Q1b. When did the biggest spikes happen?

**Approach.** Rank each commodity's monthly changes and keep the top 2. A window function can't be filtered in `WHERE` (it is evaluated after `WHERE`), so the ranking is computed in a CTE first and filtered afterwards.

*SQL: `RANK()`, filtering window results via a CTE, `NULLS LAST`.*

In [26]:
run_sql("sql/01b_lonjakan_terbesar.sql")

,komoditas,bulan,pct_mom
0,bawang_merah,2024-04-01,32.2
1,bawang_merah,2024-11-01,24.9
2,bawang_putih,2023-05-01,10.4
3,bawang_putih,2023-03-01,9.5
4,beras,2024-02-01,3.9
5,beras,2024-03-01,3.8
6,cabai_merah,2022-06-01,42.2
7,cabai_merah,2025-01-01,40.7
8,cabai_rawit,2025-12-01,58.2
9,cabai_rawit,2022-06-01,55.1


**Findings**
- Minyak goreng's only big spikes are **March–April 2022**, matching the 2022 cooking-oil crisis. This is a one-off event, not a recurring pattern.
- Chili spikes (Jun 2022, Jan 2025, Dec 2025) do **not** fall in Lebaran months. A plausible driver is seasonal/weather disruption to harvests. *This is a hypothesis only; the dataset has no weather data.*

### Q2. How much do prices rise before Lebaran?

**Approach.** For each Lebaran 2022–2025, compare the national average price in the **4 weeks before Lebaran (H-28 to H-1)** with a **baseline 2 months earlier (H-90 to H-61)**, before Ramadan.

*SQL: `VALUES` as an inline table, `CROSS JOIN`, conditional aggregation with `AVG(...) FILTER (WHERE ...)`, `AVG() OVER (PARTITION BY ...)`.*

In [27]:
run_sql("sql/02_efek_lebaran.sql")

,komoditas,tahun,kenaikan_pct,rata_4_tahun_pct
0,minyak_goreng,2022,29.9,8.3
1,minyak_goreng,2023,0.3,8.3
2,minyak_goreng,2024,1.6,8.3
3,minyak_goreng,2025,1.5,8.3
4,bawang_putih,2022,7.8,7.6
5,bawang_putih,2023,12.9,7.6
6,bawang_putih,2024,6.0,7.6
7,bawang_putih,2025,3.5,7.6
8,cabai_merah,2022,15.1,4.7
9,cabai_merah,2023,2.4,4.7


Minyak goreng looks like the most Lebaran-sensitive commodity on average, but that average is driven entirely by **2022 (+29.9%)**, the crisis year found in Q1b. A more robust summary is needed.

### Q2b. Which Lebaran effects are reliable?

**Approach.** Summarise per commodity with the **median** (resistant to one extreme year) and **how many of the 4 years** prices rose.

*SQL: `PERCENTILE_CONT(0.5) WITHIN GROUP`, `COUNT(*) FILTER (WHERE ...)`.*

In [28]:
run_sql("sql/02b_efek_lebaran_ringkas.sql")

,komoditas,median_pct,tahun_naik,terendah_pct,tertinggi_pct
0,bawang_putih,6.9,4,3.5,12.9
1,gula_pasir,2.4,4,0.4,3.6
2,minyak_goreng,1.5,4,0.3,29.9
3,daging_sapi,1.2,4,1.0,5.5
4,telur_ayam,4.5,3,-2.0,11.2
5,daging_ayam,2.7,3,-5.8,7.2
6,cabai_merah,1.9,3,-0.3,15.1
7,beras,1.9,3,0.0,6.8
8,cabai_rawit,4.3,2,-9.2,12.1
9,bawang_merah,-0.4,2,-12.2,6.3


**Findings**
- **Bawang putih** is the clearest Lebaran signal: prices rose before Lebaran in **4 of 4 years**, median **+6.9%**.
- Gula pasir and daging sapi also rose 4 of 4 years, but only modestly (median +2.4% and +1.2%).
- The median fixes the minyak goreng distortion: **+1.5%** vs a +29.9% maximum.
- Cabai rawit and bawang merah rose in only **2 of 4 years**, so there is no reliable Lebaran pattern. Their volatility (Q1) comes from elsewhere.

### Q3. Which provinces are consistently most expensive?

**Approach.** For 2025 (latest full year), express each province's price as an **index vs the national daily average** (100 = national average), then average across commodities. An index is used instead of rupiah so that expensive items (beef ≈ Rp130k/kg) don't dominate cheap ones (sugar ≈ Rp18k/kg).

*SQL: `JOIN ... USING`, a window function applied on top of `GROUP BY`.*

In [29]:
run_sql("sql/03_disparitas_provinsi.sql")

,provinsi,indeks_rata,peringkat_termahal
0,Papua,127.5,1
1,Maluku Utara,124.9,2
2,Maluku,118.6,3
3,Papua Barat,118.1,4
4,Kalimantan Utara,115.1,5
5,Kalimantan Tengah,110.1,6
6,Nusa Tenggara Timur,105.6,7
7,Sulawesi Tenggara,105.0,8
8,DKI Jakarta,103.7,9
9,Gorontalo,103.5,10


**Findings**
- Eastern Indonesia is the most expensive: Papua is **27.5% above** the national average, followed by Maluku Utara and Maluku. This is consistent with higher logistics costs to island regions.
- **Within Java**, DKI Jakarta is the most expensive (index 103.7) and **Jawa Timur the cheapest (85.7)**, second-cheapest nationally.

### Q3b. Within Java, where is each commodity cheapest?

**Approach.** Average 2025 price per commodity for the six Java provinces, ranked cheapest first, with the gap vs Jakarta. The Jakarta price is attached to every row with a filtered window aggregate, so no self-join is needed.

*SQL: `RANK() OVER (PARTITION BY ...)`, `MAX(...) FILTER (WHERE ...) OVER (...)`.*

In [30]:
run_sql("sql/03b_sourcing_jawa.sql")

,komoditas,provinsi,harga_rata,peringkat_termurah,selisih_vs_jakarta_pct
0,bawang_merah,Jawa Timur,41053.0,1,-17.0
1,bawang_merah,DI Yogyakarta,42396.0,2,-14.3
2,bawang_merah,Jawa Tengah,42809.0,3,-13.5
3,bawang_merah,Banten,45142.0,4,-8.7
4,bawang_merah,Jawa Barat,45250.0,5,-8.5
5,bawang_merah,DKI Jakarta,49470.0,6,0.0
6,bawang_putih,Jawa Timur,33979.0,1,-31.5
7,bawang_putih,Jawa Tengah,39360.0,2,-20.6
8,bawang_putih,Banten,41050.0,3,-17.2
9,bawang_putih,DI Yogyakarta,42334.0,4,-14.7


**Findings**
- **Jawa Timur is cheapest for 8 of 10 commodities** (exceptions: beras → DI Yogyakarta, minyak goreng → Banten).
- The largest gaps vs Jakarta are in the **most volatile items**: cabai rawit **−36.7%**, cabai merah **−33.0%**, bawang putih **−31.5%**.
- For eggs, oil and chicken the gap is under ~10%, and Jakarta is not always the most expensive (chicken is pricier in Banten; beef is pricier in Jawa Barat and DI Yogyakarta). Sourcing further away doesn't pay off for these.

### Q4. Long-term trend (for budgeting)

**Approach.** Yearly national average per commodity, year-over-year change (`LAG`) and cumulative change since 2022 (`FIRST_VALUE`). 2026 is excluded as incomplete.

*SQL: `LAG`, `FIRST_VALUE`, named `WINDOW` clause.*

In [31]:
run_sql("sql/04_tren_tahunan.sql")

,komoditas,tahun,harga_rata,yoy_pct,vs_2022_pct
0,bawang_merah,2022,39992.0,NaN,0.0
1,bawang_merah,2023,35926.0,-10.2,-10.2
2,bawang_merah,2024,38947.0,8.4,-2.6
3,bawang_merah,2025,45476.0,16.8,13.7
4,bawang_putih,2022,29866.0,NaN,0.0
5,bawang_putih,2023,36873.0,23.5,23.5
6,bawang_putih,2024,43158.0,17.0,44.5
7,bawang_putih,2025,42453.0,-1.6,42.1
8,beras,2022,11939.0,NaN,0.0
9,beras,2023,13734.0,15.0,15.0


**Findings**
- **Trend risk confirmed:** vs 2022, bawang putih **+42.1%**, beras **+30.5%**, gula pasir **+26.2%**, despite low monthly volatility for beras and gula.
- Beras increases are **slowing** (YoY +15.0% → +11.8% → +1.5%).
- Proteins and oil are broadly flat (daging sapi +3.9%, daging ayam +2.7%, minyak goreng back to its 2022 level).

## 7. Recommendations & limitations

### Recommendations for the client

| Group | Evidence | Recommendation |
|---|---|---|
| **Bawang putih, beras, gula** (*trend risk*) | +26–42% since 2022; bawang putih rises before Lebaran every year (median +6.9%); all three store well | **Lock in prices early**: annual supply contracts, and buy bawang putih 1–2 months before Lebaran |
| **Cabai & bawang merah** (*volatility risk*) | 11–14% average monthly swings, up to 58%; no reliable Lebaran pattern; 30–37% cheaper in Jatim/Jateng than Jakarta | **Manage volatility, don't stockpile** (perishable): backup suppliers, flexible menu pricing/portions, and a **sourcing pilot** from Jawa Timur / Jawa Tengah |
| **Proteins, eggs, oil** (*low risk*) | Low volatility, small regional gaps, flat long-term trend | Keep local suppliers; focus management attention on the two groups above |

### Limitations
- **Pre-imputed source data** (method unknown): true volatility may be higher than measured.
- **Retail traditional-market prices**, not wholesale; **logistics costs are not included**, so the Jatim–Jakarta gap is an opportunity to validate, not a guaranteed saving.
- National and regional averages are **unweighted** (each province counts equally).
- Weather/harvest explanations for chili spikes are **hypotheses**; the dataset has no weather data.
- Lebaran windows (H-28, H-90 baseline) are analytical choices; other windows could shift the magnitudes.

---

In [34]:
with engine.begin() as conn:
    conn.exec_driver_sql((ROOT / "sql" / "views.sql").read_text())

pd.read_sql("SELECT * FROM v_efek_lebaran ORDER BY tahun_naik DESC, median_pct DESC", engine)

,komoditas,median_pct,tahun_naik
0,bawang_putih,6.9,4
1,gula_pasir,2.4,4
2,minyak_goreng,1.5,4
3,daging_sapi,1.2,4
4,telur_ayam,4.5,3
5,daging_ayam,2.7,3
6,cabai_merah,1.9,3
7,beras,1.9,3
8,cabai_rawit,4.3,2
9,bawang_merah,-0.4,2


In [35]:
pd.read_sql("""
    SELECT table_name
    FROM information_schema.views
    WHERE table_schema = 'public'
    ORDER BY table_name
""", engine)

,table_name
0,v_efek_lebaran
1,v_harga_bulanan
2,v_indeks_provinsi
3,v_kenaikan_lebaran
4,v_sourcing_jawa
5,v_tren_tahunan
6,v_volatilitas


## 8. (Optional) Publish to Neon for Power BI

Power BI runs on Windows, while this database lives in a local Docker container. The table and the dashboard views (`sql/views.sql`) are therefore also published to a cloud Postgres (Neon).

In [37]:
from dotenv import load_dotenv

load_dotenv(ROOT / ".env")                       # baca .env dari root project
neon = create_engine(os.environ["NEON_URL"])     # os sudah di-import di bagian 1

harga.to_sql("harga_pangan", neon, if_exists="replace", index=False, chunksize=10_000, method="multi")

with neon.begin() as conn:
    conn.exec_driver_sql("ALTER TABLE harga_pangan ALTER COLUMN tanggal TYPE date")
    conn.exec_driver_sql("ALTER TABLE harga_pangan ADD PRIMARY KEY (tanggal, provinsi, komoditas)")
    conn.exec_driver_sql((ROOT / "sql" / "views.sql").read_text())

pd.read_sql("SELECT * FROM v_efek_lebaran ORDER BY tahun_naik DESC, median_pct DESC", neon)

,komoditas,median_pct,tahun_naik
0,bawang_putih,6.9,4
1,gula_pasir,2.4,4
2,minyak_goreng,1.5,4
3,daging_sapi,1.2,4
4,telur_ayam,4.5,3
5,daging_ayam,2.7,3
6,cabai_merah,1.9,3
7,beras,1.9,3
8,cabai_rawit,4.3,2
9,bawang_merah,-0.4,2


In [39]:
for eng in (engine, neon):
    with eng.begin() as conn:
        conn.exec_driver_sql((ROOT / "sql" / "views.sql").read_text())

pd.read_sql("SELECT * FROM v_matriks_risiko ORDER BY rata_gejolak_bulanan_pct DESC", neon)

,komoditas,rata_gejolak_bulanan_pct,kenaikan_2022_2025_pct,efek_lebaran_median_pct,lebaran_tahun_naik
0,cabai_rawit,14.3,3.7,4.3,2
1,cabai_merah,11.7,5.1,1.9,3
2,bawang_merah,11.2,13.7,-0.4,2
3,bawang_putih,3.1,42.1,6.9,4
4,daging_ayam,2.7,2.7,2.7,3
5,telur_ayam,2.5,8.5,4.5,3
6,minyak_goreng,1.6,0.1,1.5,4
7,beras,0.9,30.5,1.9,3
8,gula_pasir,0.9,26.2,2.4,4
9,daging_sapi,0.4,3.9,1.2,4
